src/lib-test/fuzzer.h

```diff
@@ -0,0 +1,16 @@
+#ifndef FUZZER_H
+#define FUZZER_H
+
+#define FUZZ_DATA(args) \
+	int LLVMFuzzerTestOneInput(args); \
+	int LLVMFuzzerTestOneInput(args) {
+
+#define FUZZ_STR(str_arg) \
+	int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size); \
+	int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) { \
+		str_arg = t_strndup(data, size);
```

Comment:
Should this not be freed ?

Reply:
t_stuff do  not need to be explicitly freed.

Reply:
Is LLVMFuzzerTestOneInput running in a loop? I didn't find out how exactly it was called, so I thought maybe it recreates the process between runs. But I guess not?

Reply:
Added T_BEGIN .. T_END

Reply:
> t_stuff do not need to be explicitly freed

Oh... How does it work ?

> Is LLVMFuzzerTestOneInput running in a loop?

Usually yes cf libFuzzer. But you can write a simple driver that runs it only once, or that runs it for all the files in a specific directory...